core/variant/variant_call.cpp

```diff
@@ -753,40 +753,56 @@ struct _VariantCall {
 	static PackedInt32Array func_PackedByteArray_decode_s32_array(PackedByteArray *p_instance) {
 		uint64_t size = p_instance->size();
 		PackedInt32Array dest;
-		ERR_FAIL_COND_V_MSG(size < sizeof(int32_t), dest, "Size didn't match array of size int32_t, maybe you are trying to convert to the wrong type?");
+		if (size == 0) {
+			return dest;
+		}
+		static_assert(sizeof(dest[0]) == sizeof(int32_t));
```

Comment:
I'm not familiar with this, isn't that indexing out of bounds since `dest` is empty here? Or this is just getting the address of the would-be first element?

Reply:
Should be fine as `dest[0]` won't be evaluated [as an expression in `sizeof`](https://en.cppreference.com/w/cpp/language/expressions#Unevaluated_expressions). It will only be deduced at the compile-time what the return type of `dest[0]` is. And currently it will deduce it's `int32_t` because that's what this operator returns:
https://github.com/godotengine/godot/blob/f6072c9c98f11797e31a2076d17b580a6c651679/core/variant/variant_call.cpp#L755
https://github.com/godotengine/godot/blob/f6072c9c98f11797e31a2076d17b580a6c651679/core/variant/variant.h#L65
https://github.com/godotengine/godot/blob/f6072c9c98f11797e31a2076d17b580a6c651679/core/templates/vector.h#L92

Reply:
I know it's redundant. I can remove it. I added it while debugging since I was trying to figure out the problem, and it seemed harmless (no runtime overhead) so I kept it.

Reply:
Yeah if there's no way this can fail, it's probably best to remove it. Otherwise there are gazillion places in the codebase where such asserts could be used to check that a strong typed variable is its type, which wouldn't help much.